pkg/trie/triedb/iterator.go

```diff
@@ -6,11 +6,38 @@ package triedb
 // Entries returns all the key-value pairs in the trie as a map of keys to values
 // where the keys are encoded in Little Endian.
 func (t *TrieDB) Entries() (keyValueMap map[string][]byte) {
-	panic("not implemented yet")
+	entries := make(map[string][]byte)
+
+	iter := NewTrieDBIterator(t)
+	for entry := iter.NextEntry(); entry != nil; entry = iter.NextEntry() {
+		entries[string(entry.key)] = entry.value
+	}
+
+	return entries
 }
 
 // NextKey returns the next key in the trie in lexicographic order.
 // It returns nil if no next key is found.
 func (t *TrieDB) NextKey(key []byte) []byte {
-	panic("not implemented yet")
+	iter := NewTrieDBIterator(t)
+
+	// TODO: Seek will potentially skip a lot of keys, we need to find a way to
```

Comment:
Do you plan to address this TODO in this PR?

Reply:
I'll create an issue and address it after having the writes working, I want to focus on that to finish the overall work about the tries and then focus on these improvements. Thanks for asking